application/comit_node/src/bin/comit_node.rs

```diff
@@ -80,6 +83,7 @@ fn main() -> Result<(), failure::Error> {
         alice_protocol_dependencies,
         Arc::clone(&swarm),
         local_peer_id,
+        Arc::clone(&swarm),
```

Comment:
:wrench: Creating a new trait for each piece of functionality doesn't really scale since we are passing the `Swarm` now already two times.

What do you think about unifying `BamPeers` and `ListenAddresses` to something like `SwarmInfo` that provides us convenient access to all kinds of data from the `Swarm`?